extensions/_beacons/http_status.py

```diff
@@ -105,30 +105,36 @@ def beacon(config):
                 r = requests.get(url, timeout=sites_config['timeout'])
             else:
                 r = requests.get(url, timeout=30)
+        # r.status_code
         except requests.exceptions.RequestException as e:
-            log.debug("Request failed: ", e)
+            log.debug("Request failed: %s", e)
+        if r.status_code >= 500:
+            log.debug('Response from status endpoint was invalid: '
+                      '%s', r.status_code)
         for json_response_item in sites_config['json_response']:
-            service = json_response_item['path'].split(':')[0]
-            service_value = json_response_item['path'].split(':')[1]
+            if ':' in json_response_item['path']:
+                service = json_response_item['path'].split(':')[0]
+                service_value = json_response_item['path'].split(':')[1]
             if service in r.json():
                 if json_response_item['comp'] in comparisons:
                     comp = comparisons[json_response_item['comp']]
                     if not comp(json_response_item['value'],
                                 r.json()[service][service_value]):
                         _failed = {'service': service,
                                    'status': json_response_item['value'],
-                                   'comp': comp,
+                                   'comp': json_response_item['comp'],
                                    }
                         ret.append(_failed)
                 else:
                     log.debug('Comparison operator not in comparisons dict: '
-                              ' %s', json_response_item['value'])
+                              '%s', json_response_item['value'])
             else:
                 log.debug('Server status response does not include listed '
                           'service in path: %s', service)
-        for html_response_item in sites_config['html_response']:
-            search_value = html_response_item['value']
-            comp = comparisons[html_response_item['comp']]
-            if not comp(search_value, r.text):
-                ret.append({'keyword': search_value})
+        if 'html_response' in sites_config:
```

Comment:
Rather than an `if` clause you could do
```suggestion
        for html_response_item in sites_config.get('html_response', []):
```